emacs-ide.md

```diff
@@ -247,6 +247,16 @@ To **evaluate** rather than compile:
 - evaluate the sexp before the point with `C-x C-e`.
 See also other commands in the menu.
 
+---
+**EVALUATION VS COMPILATION**
+
+There are a couple of pragmatic differences when choosing between compiling or evaluating.
+In general, it is better to *compile* top-level forms, for two reasons:
+* Compiling a top-level form highlights warnings and errors in the editor, evaluation does not.
+* SLIME keeps track of line-numbers of compiled forms; when a top-level form is evaluated, the file line number information is lost.
+That's problematic for code navigation afterwards.
+---
+
```

Comment:
An example when sometimes you prefer to eval?

Reply:
To compile any defun or class. Have you tried this yourself? Perhaps it is better if you try it and see what I mean.

Say you have this function in your file:

```lisp
(defun foo (x))
```
Go to a function and evaluate it instead of compiling it (C-c C-e). You won't see any warnings highlighted in source, only warnings in repl.

After that, try navigating to FOO using M-. . You want be able to (slime-edit-definition-cont: Error: DEFINITION-SOURCE of function FOO did not contain meaningful information.)

Now compile it instead (C-c C-c). You will see the warning highlighted in editor.  And try navigate to it with M-. . Success!

Reply:
OK. And in the contrary, are there some situations where you don't compile, but prefer to eval? (that would be to illustrate further what your wrote, suggesting that eval has its use)

Reply:
Use eval to observe results from non top-level forms.

Say you have this function:

```
(defun foo ()
  (let ((f (open "/home/marian/test.lisp")))
    ...))
```
Go to the end of the OPEN expression and evaluate it (C-x C-e), to observe the result:

```
=> #<SB-SYS:FD-STREAM for "file /mnt/e6b00b8f-9dad-4bf4-bd40-34b1e6d31f0a/home/marian/test.lisp" {1003AAAB53}>
```

Reply:
oh, indeed. I think that would make a great addition :) 

(I don't use that much myself, I'll try)

Reply:
I think that Evaluation appearing before Compilation in the SLIME manual is unfortunate.

Because when you are learning Lisp and SLIME, you learn to evaluate before learning to compile, and may never see the difference.

That's why I'm trying to highlighting the difference in the cookbook.

Reply:
These things are easy to miss just reading a manual. Perhaps it is better to highlight them in a video. But anyway ...